### DISCUSSION TITLE
API change - Override settings not working

### DISCUSSION THREAD

**Discussion by Negatrev** [lllyasviel/stable-diffusion-webui-forge#1026] [Q&A]
![image](https://github.com/user-attachments/assets/d15b7054-78da-4930-8476-5e0e2da24f3c)

So before I recently updated I could use this option in the text2iage payload to change the model alongside the request. But now I'm not getting any error message, every other option is applying, but the model is no longer changing. Does anyone else us this option and having the same problem? Is there a new FastAPI being used and the docs just aren't updated?

**Comment by Hugo-Matias** [comment]
Also trying to load a model using the API. Migrated over the weekend to Forge from Auto's and other than this it's been a smooth transition.

I used to POST to `/sdapi/v1/options` with just `{"sd_model_checkpoint": "some_model_name.safetensors"}` as payload, that would trigger the model loading. Doing this now seems to change the config but there is no feedback from the console, hitting the inference endpoint will keep using the loaded model.

Also tried to POST `/sdapi/v1/reload-checkpoint` after the change but no luck either.

**Reply by Negatrev** [reply]
Yes, Options is the same call as the override_settings (except the latter is called at generation). I suspect something changed in how the models are loaded through API. Either a bug or new syntax.

Ignore my python variables, but the syntax is identical for the model.

"override_settings": {"sd_model_checkpoint": get_model(style),"sd_vae": get_vae(style)},

**Reply by Hugo-Matias** [reply]
I see, it's not the workflow I use but it makes sense. 
I was able to hack a solution for the reload-checkpoint endpoint, now I'm just calling both endpoints and it seems to be working more or less as I would expect (model is still loaded before inference but that's fine by me).

`reload-checkpoint` seems to be reaching this unimplemented function.

https://github.com/lllyasviel/stable-diffusion-webui-forge/blob/f6ef105cb333692d8695ebe14c8409f98675e2f4/modules/sd_models.py#L425-L426

This is my simple solution for now. I'm probably breaking lots of conventions or probably doing it in a weird way logically but I'm not familiar with the codebase at deeper level.

```python
def send_model_to_device(m):
    from modules_forge import main_entry
    main_entry.refresh_model_loading_parameters()
    return
```

Seems to be how the main web ui selects the model so I'm just calling the same method from the api. Maybe you can use that on  txt2img/img2img handler functions and intercept the override_settings logic somehow.

**Reply by dermesut** [reply]
while this workaround seems to work for me too, i find it very frustrating that so much of the possibilities we had to communicate with the forge-code from outside, like our own apps, seems to be just gone.
i can't really be the way to go, to fiddle with the original forge-code, just to change the checkpoint one wants to use in ones app.
what about all the rest? can't change the vae, what about "hr_checkpoint_name", meaning can't change the checkpoint for hires fix.
at least not in the way, we used to. who knows, how it's done now. if at all.
it's becoming a very frustrating experience to try to adapt my app to this new version of forge.
end of rant.

**Reply by Hugo-Matias** [reply]
I'm optimistic about it. I lack the expertise to push PR's and help out at that level but having Flux adaptation so early on means lot's of people are implementing this web-ui in their workflows, that also means more talent is reaching the code base and exploring the issues that need to be ironed out like these. lllyasviel has been doing a fantastic job with the project, I'll keep using it for the time being.

**Reply by dermesut** [reply]
`lllyasviel has been doing a fantastic job with the project`
there's absolutely no doubt about that, imo.
since forge's "birth" it's been my go-to. that's the frustrating part about it.
i use this image-generating stuff solely within my own plugin that i am writing (for c4d) - but am in no way shape or form knowledgable enough to help, or ciritcize even, the code-base of forge.
quite frankly, i sh*t my pants, having to get into the code and change something in it, just to be able to, e.g., change the checkpoint i want to use. lol.
i really do hope, that the usage of forge via external code at one point becomes as "obvious" as it was.
not sure, that i can share your optimism though.
the development seems to be going steadily in the direction of adding more and more modern features, while leaving that external part alone, more or less.

**Comment by gonewilds** [comment]
use /sdapi/v1/options and post { sd_model_checkpoint: modelname.safetensors } as the payload. it work fine for me

**Reply by Negatrev** [reply]
What version are you using as this wasn't working for https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/1026#discussioncomment-10317708

PS. I know other methods exist. I was reporting that this method wasn't.

**Reply by Enferlain** [reply]
Yeah it didn't seem to work for me either. @Negatrev did you figure out anything since last week?

update: apparently it does work, forge gets the request, so that's nice
